lib/metrics/amplitude.js

```diff
@@ -105,11 +105,19 @@ const EVENT_PROPERTIES = {
   [GROUPS.sms]: NOP
 }
 
-module.exports = log => {
-  if (log) {
-    return receiveEvent
+module.exports = (log, config) => {
+  if (! log || ! Array.isArray(config.oauth.clientIds)) {
+    throw new TypeError('Bad argument')
```

Comment:
This also mops up an outstanding review comment: https://github.com/mozilla/fxa-auth-server/pull/2069#discussion_r135435005